Logzio k8s events

ralongit opened the pull request:
Added a helm chart that support sending Kubernetes deployment events to Logz.io.

ralongit committed:
Initial deployment of k8s events monitoring

ralongit committed:
Readme update and fix chart version

Fix namespace in readme, add chart to global readme & fix chart version.

ralongit committed:
Fixes template definition, format and includes

ralongit committed:
Convert to deployment, replace to env_id & add custom listener

ralongit committed:
Switch image tag from using tpl to its value

mirii1994 commented on charts/logzio-k8s-events/values.yaml:
```diff
@@ -0,0 +1,70 @@
+
+
+image:
+  repository: ralatdocker/logzio-k8s-events
```
Please change to logzio repo before merging :)

mirii1994 commented on charts/logzio-k8s-events/values.yaml:
```diff
@@ -0,0 +1,70 @@
+  tag: "0.0.1"
+
+k8sApp: logzio-k8s-events
+namespace: monitoring
```
It's better to use {{ .Release.Namespace }} instead of allowing to set it elsewhere. This way, you make sure that there's a match between the namespace of the Chart itself and the namespace of the resources it created.
You can see an example here
https://github.com/logzio/logzio-helm/blob/a0b22fdfc10c229d5f17f94b264e727bd53a2a94/charts/fluentd/templates/clusterrole.yaml#L5

mirii1994 reviewed (changes requested):
Great work, please see my comments

ralongit committed:
Update Readme file & specify release namespace for each k8s object

- Changed namespace value to: {{ .Release.Namespace }} in all resources
- Remove redundant readme text and update env id.

mirii1994 commented on charts/logzio-k8s-events/values.yaml:
```diff
@@ -4,12 +4,12 @@
 
 k8sApp: logzio-k8s-events
-namespace: monitoring
+namespace: {{ .Release.Namespace }} 
```
You don't need to specify this, the `.Release.Namespace` is sort of a a "reserved variable" of helm. You can just delete the namespace line from values.yaml.

mirii1994 commented on charts/logzio-k8s-events/templates/clusterrolebinding.yaml:
```diff
@@ -3,6 +3,7 @@ apiVersion: {{ .Values.apiVersions.clusterRoleBinding }}
 kind: ClusterRoleBinding
 metadata:
   name: {{ include "logzio-k8s-events.fullname" . }}
+  namespace: {{ .Release.Namespace }} 
```
I'm not sure if clusterrolebinding requires a namespace, please check.

ralongit replied:
It doesn't, removed.

mirii1994 commented on charts/logzio-k8s-events/templates/clusterrole.yaml:
```diff
@@ -3,7 +3,7 @@ apiVersion: {{ .Values.apiVersions.clusterRole }}
 metadata:
   name: {{ include "logzio-k8s-events.fullname" . }}
-  namespace: {{ .Values.namespace }}
+  namespace: {{ .Release.Namespace }} 
```
I'm not sure if clusterrole requires a namespace, please check.

ralongit replied:
It doesn't, removed.

ralongit committed:
Remove redundant namespace value

mirii1994 reviewed (approved):
Amazing.
Let's wait a little bit with merging.

ralongit committed:
Update clusterrolebinding service account subject namespace

Update clusterrolebinding service account subject namespace

ralongit committed:
Rename secret

Changed the value of secretName to different one in order to avoid conflicts with the unified chart secret.
